### PR TITLE
Feature/facebook log in

### PR DESCRIPTION
**Pull request by sansive**
## Task
Implement Facebook Log in

## Test
Check buttons for Facebook Log in

**Note:** Please check you can actually log in with Facebook in case Github invalidates the tokens because of the leak

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -5,9 +5,10 @@
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
-    <uses-feature android:name="android.hardware.camera.any" />
-
     <uses-permission android:name="android.permission.CAMERA" />
+    <uses-permission android:name="android.permission.INTERNET" />
+
```
Comment (aindreias) [inline app/src/main/AndroidManifest.xml:10]: Why do you need the Internet permission? Is it useful for this PR?

**Review by aindreias — CHANGES_REQUESTED** [review]
I'm not able to test the actual login behaviour (lost access to my FB account). However your code looks all good to me, left 1 comment & I'm approving once it's addressed 👍

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (sansive) [reply app/src/main/AndroidManifest.xml:10]: It was written in the developer's guide, I believe it is necessary to access your Facebook account

Comment (aindreias) [reply app/src/main/AndroidManifest.xml:10]: Alright, that makes sense - thanks!

**Review by aindreias — APPROVED** [review]
Nice work! And thank you for the explanation 👍

**Review by Geno22 — CHANGES_REQUESTED** [review]
When I sign-in and go to the main Screen it crashes and displays this error :
`E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.github.houseorganizer.houseorganizer, PID: 2753
    java.lang.RuntimeException: Failure delivering result ResultInfo{who=@android:requestPermissions:, request=99, result=-1, data=Intent { act=android.content.pm.action.REQUEST_PERMISSIONS (has extras) }} to activity {com.github.houseorganizer.houseorganizer/com.github.houseorganizer.houseorganizer.panels.main_activities.MainScreenActivity}: java.lang.NullPointerException
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5015)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:5056)
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: java.lang.NullPointerException
        at java.util.Objects.requireNonNull(Objects.java:220)
        at com.github.houseorganizer.houseorganizer.panels.main_activities.MainScreenActivity.selectHouse(MainScreenActivity.java:228)
        at com.github.houseorganizer.houseorganizer.panels.main_activities.MainScreenActivity.loadData(MainScreenActivity.java:210)
        at com.github.houseorganizer.houseorganizer.panels.main_activities.MainScreenActivity.onRequestPermissionsResult(MainScreenActivity.java:201)
        at android.app.Activity.dispatchRequestPermissionsResult(Activity.java:8466)
        at android.app.Activity.dispatchActivityResult(Activity.java:8314)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5008)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:5056) 
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:223) 
        at android.app.ActivityThread.main(ActivityThread.java:7656) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) `
I tried multiple times but it keeps crashing
I am pretty sure it is nothing but I don't know where it is from

**Comment by Geno22** [comment]
Yes I did

**Review by Geno22 — APPROVED** [review]
Great work thank for the changes !

**Pull request merged**